### DISCUSSION TITLE
Forward was compeleted successfully, but the HTLC is still pending

### DISCUSSION THREAD

**Discussion by SunnySarahNode** [lightningnetwork/lnd#8511] [General]
I'm seeing a very weird situation that I haven't encountered before, and which seems like it could lead to loss of funds.
I hope I just misunderstand something and it's not a bug :)

Me: LND 0.17.3, bitcoind 25 (full indexed blockchain), Ubuntu (amd64), clearnet only
My peer: unknown

I have the following channel with the HTLC in it (` lncli listchannels` ) :
```

{
            "active": false,
            "remote_pubkey": "0383a52db8e4063714c767e90628a5caee4d278c308cfe0fae82e21aebc313ccbe",
            "channel_point": "db8486422565a803e8dc82b4424e3ecadb4aa5cc942284079ede4a91a453fe98:1",
            "chan_id": "877767620541415425",
            "capacity": "4000000",
            "local_balance": "684572",
            "remote_balance": "532248",
            "commit_fee": "3574",
            "commit_weight": "1288",
            "fee_per_kw": "2757",
            "unsettled_balance": "2778946",
			...
            "pending_htlcs": [
                {
                    "incoming": true,
                    "amount": "2778946",
                    "hash_lock": "4ac9b86355ffed94847b01017d1f5976658883c75bb35a0a72fbb22cdee7ebdb",
                    "expiration_height": 832737,
                    "htlc_index": "551934",
                    "forwarding_channel": "887966690141143041",
                    "forwarding_htlc_index": "10105"
                }
            ],
            "csv_delay": 480,
            "chan_status_flags": "ChanStatusDefault",
            "local_chan_reserve_sat": "40000",
            "remote_chan_reserve_sat": "40000",
            "static_remote_key": false,
            "commitment_type": "ANCHORS",
            "local_constraints": {
                "csv_delay": 480,
                "chan_reserve_sat": "40000",
                "dust_limit_sat": "354",
                "max_pending_amt_msat": "3960000000",
                "min_htlc_msat": "1",
                "max_accepted_htlcs": 483
            },
            "remote_constraints": {
                "csv_delay": 480,
                "chan_reserve_sat": "40000",
                "dust_limit_sat": "354",
                "max_pending_amt_msat": "3960000000",
                "min_htlc_msat": "100000",
                "max_accepted_htlcs": 483
            },
            "peer_alias": "blitzbold",
        }

```

This forward has the following route :

blitzbold --> (me) --> LOOP

This forward was successfully completed ( `lncli fwdinghistory` ) :

```
        {
            "timestamp": "1709147676",
            "chan_id_in": "877767620541415425",
            "chan_id_out": "887966690141143041",
            "amt_in": "2778946",
            "amt_out": "2773546",
            "fee": "5400",
            "fee_msat": "5400094",
            "amt_in_msat": "2778946094",
            "amt_out_msat": "2773546000",
            "timestamp_ns": "1709147676006760972",
            "peer_alias_in": "blitzbold",
            "peer_alias_out": "LOOP"
        }
```

...but as you can see, the HTLC in the channel 877767620541415425 is still "pending".

My peer is unable to receive connections correctly, so this channel is "disabled" :

```
2024-02-28 23:33:38.435 [ERR] RPCS: [/lnrpc.Lightning/DisconnectPeer]: unable to disconnect peer: peer 0383a52db8e4063714c767e90628a5caee4d278c308cfe0fae82e21aebc313ccbe is not connected
2024-02-28 23:33:53.877 [INF] SRVR: Established connection to: 0383a52db8e4063714c767e90628a5caee4d278c308cfe0fae82e21aebc313ccbe@46.101.122.181:49621
2024-02-28 23:33:53.877 [INF] SRVR: Finalizing connection to 0383a52db8e4063714c767e90628a5caee4d278c308cfe0fae82e21aebc313ccbe@46.101.122.181:49621, inbound=false
2024-02-28 23:33:53.967 [INF] PEER: Peer(0383a52db8e4063714c767e90628a5caee4d278c308cfe0fae82e21aebc313ccbe): disconnecting 0383a52db8e4063714c767e90628a5caee4d278c308cfe0fae82e21aebc313ccbe@46.101.122.181:49621, reason: unable to start peer: unable to read init msg: read next header: read tcp [ip]:34816->46.101.122.181:49621: read: connection reset by peer

```

So, for now :

1) My local liquidity in the channel with LOOP was decreased.
2) My local liquidity in the channel with blitzbold was NOT increased.
3) HTLC in the channel with blitzbold is "pending", channel is "disabled".

The main question : what exactly will happen at block 832737, when this channel is auto-FCed due to expiration of this HTLC? Will my peer receive funds (this is a large amount!) that should belong to me, because this is an _outgoing_ HTLC for him?

**Comment by bitromortac** [comment]
This seems to be a normal case of an HTLC that can't be settled because the peer isn't reachable. That's covered in this test: 
https://github.com/lightningnetwork/lnd/blob/72764b14734870d03a1212795e2ec7c02f3b4500/itest/lnd_multi-hop_test.go#L338-L547
so I think your node should go to chain to claim the HTLC 10 blocks before timeout, see https://github.com/lightningnetwork/lnd/blob/72764b14734870d03a1212795e2ec7c02f3b4500/lncfg/config.go#L20-L27

**Comment by SunnySarahNode** [comment]
[bitromortac](https://github.com/bitromortac) Thank you for your answer, it's really helpful!

One more short question: can I FC this channel manually earlier with the same result?

**Reply by bitromortac** [reply]
I think that lnd should handle it, but there's a chance that the peer comes back online? It may also be that you provoke an edge case, it's maybe safer to let lnd go through standard procedures, but not sure.
